### PR TITLE
Fix a variety of doxygen problems.

### LINKED ISSUE #1273: there is a 404 documentation page
Access to 
https://www.dealii.org/developer/doxygen/deal.II/classDerivativeForm.html
will have a 404 error. 

I am not sure wether this is a problem or not. But I checked the file
```
 dealii/include/deal.II/base/derivative_form.h
```
still exists.

### PR DESCRIPTION
**Pull request by bangerth**
Fixes #1273.

### CODE REVIEW

File: doc/doxygen/CMakeLists.txt [hunk 1]
```diff
@@ -104,7 +104,6 @@ LIST(APPEND _doxygen_input
 SET(_doxygen_image_path
   ${CMAKE_CURRENT_SOURCE_DIR}/images
   ${CMAKE_SOURCE_DIR}/contrib/parameter_gui/images
-  ${CMAKE_CURRENT_BINARY_DIR}/images
```
Comment (tjhei) [inline doc/doxygen/CMakeLists.txt]: I am not sure, but isn't this the path ```contrib/utilities/makeofflinedoc.sh``` dumps the images into?

**Comment by bangerth** [comment]
Good point. I've dropped the directory only if it doesn't exist.

**Comment by tjhei** [comment]
I'm not sure this works as expected. For example in the release process we do "make documentation; make install; cd ../installed/doc; ../../BLA/make_offline_doc.sh". Maybe it doesn't matter though. What is the doxygen image path used for anyway? We download the images and replace the URLs so maybe your first version that just doesn't include it is correct after all...

**Comment by bangerth** [comment]
When you reference images from `@image` statements, doxygen looks for them in a list of directories and copied them into the output directory so that *all* of the output is in one place. My reading of the script is that it creates this directory not because it wants doxygen to look through it, but in fact *after* doxygen has already run -- `makeofflinedoc.sh` doesn't actually call `doxygen` -- and in fact that is what I see in `homepage/RELEASE/make_offline_doc.sh`:
```
$PATCH

# Generate documentation
rm -rf builddoc
mkdir builddoc
cd builddoc
cmake -DDEAL_II_WITH_MPI=OFF -DCMAKE_BUILD_TYPE=Debug -DDEAL_II_COMPONENT_DOCUMENTATION=ON -DCMAKE_INSTALL_PREFIX=`pwd`/../installeddoc/ ../
make documentation -j 8

cd ../installeddoc

cd doc/doxygen/deal.II
../../../../contrib/utilities/makeofflinedoc.sh
```
In other words, I think the original patch was correct all along. Would you agree?

**Comment by tjhei** [comment]
Yes, we create and fill the images/ path after doxygen is run, so your original version should work (deleting images/ from the doxygen paths).

**Comment by bangerth** [comment]
OK, I went back to the original version.

**Comment by tjhei** [comment]
Sorry for the back and forth on this. :-)
